modules/more_pickers.js

```diff
@@ -0,0 +1,198 @@
+/* see more_pickers.md for more information */
+
+exports.doublePicker = function (options) {
+  var menuIcon = "\0\f\f\x81\0\xFF\xFF\xFF\0\0\0\0\x0F\xFF\xFF\xF0\0\0\0\0\xFF\xFF\xFF";
+
+  var R = Bangle.appRect;
+  g.reset().clearRect(R);
+  g.setFont("12x20").setFontAlign(0, 0).drawString(menuIcon + " " + options.title, R.x + R.w / 2, R.y + 12);
+
+  var v_1 = options.value_1;
+  var v_2 = options.value_2;
+
+  function draw() {
+    g.setColor(g.theme.bg2)
+      .fillRect(14, 60, 81, 166)
+      .fillRect(95, 60, 162, 166);
+
+    g.setColor(g.theme.fg2)
+      .fillPoly([47.5, 68, 62.5, 83, 32.5, 83])
+      .fillPoly([47.5, 158, 62.5, 143, 32.5, 143])
+      .fillPoly([128.5, 68, 143.5, 83, 113.5, 83])
+      .fillPoly([128.5, 158, 143.5, 143, 113.5, 143]);
+
+    var txt_1 = options.format_1 ? options.format_1(v_1) : v_1;
+    var txt_2 = options.format_2 ? options.format_2(v_2) : v_2;
+
+    g.setFontAlign(0, 0)
+      .setFontVector(Math.min(30, (R.w - 110) * 100 / g.setFontVector(100).stringWidth(txt_1)))
+      .drawString(txt_1, 47.5, 113)
+      .setFontVector(Math.min(30, (R.w - 110) * 100 / g.setFontVector(100).stringWidth(txt_2)))
+      .drawString(txt_2, 128.5, 113)
+      .setFontVector(30)
+      .drawString(options.separator ?? "", 88, 110);
```

Comment:
@gfwilliams looks like the IDE is out of sync with espruino:

![image](https://github.com/espruino/BangleApps/assets/205673/0503a4a0-c579-459c-9a2d-9fc7671e2e10)

`??` is parsed and handled fine

Reply:
Yes - it uses Acorn and Tern for parsing and type inference, but I guess they are older versions that don't support ??